File: parse_pdf.py
import fitz  # PyMuPDF
import sys

def parse_pdf(file_path, output_path):
    try:
        doc = fitz.open(file_path)
        text = ""
        for page in doc:
            text += page.get_text()
        with open(output_path, "w", encoding="utf-8") as f:
            f.write(text)
        print("PDF extracted successfully!")
    except Exception as e:
        print(f"Error: {e}")

if __name__ == "__main__":
    parse_pdf("knave2翻译文档.pdf", "knave2-text.txt")

File: parse_pdf.mjs
import fs from 'fs';
import pdf from 'pdf-parse/lib/pdf-parse.js';

let dataBuffer = fs.readFileSync('knave2翻译文档.pdf');

pdf(dataBuffer).then(function (data) {
    fs.writeFileSync('knave2-text.txt', data.text);
    console.log("PDF text extracted successfully.");
}).catch(function (err) {
    console.error("Error extracting text: ", err);
});
